app/src/main/java/com/duckduckgo/app/autocomplete/api/AutoComplete.kt

```diff
@@ -134,33 +138,30 @@ class AutoCompleteApi @Inject constructor(
     private val autocompleteTabsFeature: AutocompleteTabsFeature,
 ) : AutoComplete {
 
-    override fun autoComplete(query: String): Observable<AutoCompleteResult> {
+    private var isAutocompleteTabsFeatureEnabled: Boolean? = null
+
+    override fun autoComplete(query: String): Flow<AutoCompleteResult> = flow {
         if (query.isBlank()) {
-            return Observable.just(AutoCompleteResult(query = query, suggestions = emptyList()))
+            return@flow emit(AutoCompleteResult(query = query, suggestions = emptyList()))
         }
-        val savedSitesObservable: Observable<List<AutoCompleteSuggestion>> =
-            getAutoCompleteBookmarkResults(query)
-                .zipWith(
-                    getAutoCompleteFavoritesResults(query),
-                ) { bookmarks, favorites ->
-                    (favorites + bookmarks.filter { favorites.none { favorite -> (it.suggestion).url == favorite.suggestion.url } })
-                }.zipWith(
-                    getAutocompleteSwitchToTabResults(query),
-                ) { bookmarksAndFavorites, tabs ->
-                    (tabs + bookmarksAndFavorites) as List<RankedSuggestion<AutoCompleteUrlSuggestion>>
-                }.zipWith(
-                    getHistoryResults(query),
-                ) { bookmarksAndFavoritesAndTabs, historyItems ->
-                    val searchHistory = historyItems.filter { it.suggestion is AutoCompleteHistorySearchSuggestion }
-                    val navigationHistory = historyItems
-                        .filter { it.suggestion is AutoCompleteHistorySuggestion } as List<RankedSuggestion<AutoCompleteHistorySuggestion>>
-                    (removeDuplicates(navigationHistory, bookmarksAndFavoritesAndTabs) + searchHistory)
-                        .sortedByDescending { it.score }
-                        .map { it.suggestion }
-                }
+        val savedSites = getAutoCompleteBookmarkResults(query)
```

Comment:
For simplicity, can we just combine all flows in one single operation?

Reply:
Thanks for the suggestion :+1: Updated.